distributed/protocol/arrow.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import print_function, division, absolute_import
+
+from .serialize import register_serialization
+
+
+def serialize_batch(batch):
+    import pyarrow as pa
+    sink = pa.BufferOutputStream()
+    writer = pa.RecordBatchStreamWriter(sink, batch.schema)
+    writer.write_batch(batch)
+    writer.close()
```

Comment:
One improvement on the arrow side would be if `RecordBatchStreamWriter` was a context manager as that would avoid the need for an explicit close.

Reply:
A fair point. ARROW-2863